kernel/mem_domain.c

```diff
@@ -164,13 +164,18 @@ void k_mem_domain_add_thread(struct k_mem_domain *domain, k_tid_t thread)
 void k_mem_domain_remove_thread(k_tid_t thread)
 {
 	unsigned int key;
+	struct k_mem_domain *domain;
 
 	__ASSERT(thread && thread->mem_domain_info.mem_domain, "");
 
 	key = irq_lock();
+	domain = thread->mem_domain_info.mem_domain;
+	if (_current == thread) {
+		_arch_mem_domain_destroy(domain);
```

Comment:
What happens if other threads were members of the domain?

Reply:
Other threads doesn't matter since this will just set the page properties for the current thread. This arch specific destroy doesn't manipulate the mem_domain struct. So for the other threads which are a part of this mem domain will be configured when they get swapped in.